Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -983,7 +985,8 @@ class CombinatorialKalmanFilter {
       result.fittedStates.applyBackwards(lastMeasurementIndex, [&](auto st) {
         bool isMeasurement =
             st.typeFlags().test(TrackStateFlag::MeasurementFlag);
-        if (isMeasurement) {
+        bool isMaterial = st.typeFlags().test(TrackStateFlag::MaterialFlag);
+        if (isMeasurement || isMaterial) {
           firstMeasurementIndex = st.index();
```

Comment:
First "measurement" is a bit of a misnomer then in case it's a material surface, right?

Reply:
Yes you are right I didn't think to change it. I could use replace `firstMeasurementIndex` by `firstStateIndex` and `firstCreatedMeasurement` by `firstCreatedState`